src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,56 @@
+package duke.task;
+
+import duke.DukeException;
+import duke.util.DateTimeHandler;
+
+/**
+ * Represents a deadline as a task with a date and time.
+ */
+public class Deadline extends Task {
+
+    /** Date and time of the deadline. */
+    private final String dateTime;
+
+    /**
+     * Creates a new deadline from a description and a date and time String.
+     *
+     * @param description Description of the deadline.
+     * @param by Due date and time of the deadline.
+     * @throws DukeException If the date and time input is invalid.
+     */
+    public Deadline(String description, String by) throws DukeException {
```

Comment:
Perhaps a more descriptive term than `by` can be used to describe the deadline.

Reply:
Changed to dateTime.